### ISSUE TITLE
[Bug]: ax.invert_xaxis() and ax.invert_yaxis() both flip the X axis

### ISSUE THREAD

**Issue by Mars-or-bust** [matplotlib/matplotlib#21369]
### Bug summary

The ax.invertxaxis() and ax.invert_yaxis() function both produce the same output, a scatterplot with a flipped X axis. 

### Code for reproduction

```python
from mpl_toolkits import mplot3d
import matplotlib.pyplot as plt

ax = plt.axes(projection='3d')
plt.title("Invert Z")
ax.scatter3D(1,1,1)
# ax.invert_xaxis()
ax.invert_yaxis()
# ax.invert_zaxis()
```

### Actual outcome

No inversions: 
![Screen Shot 2021-10-15 at 1 52 25 PM](https://user-images.githubusercontent.com/35809470/137531316-0b630f80-6077-436e-9af3-78e473c634b5.png)

X Inversion: 
![Screen Shot 2021-10-15 at 1 52 48 PM](https://user-images.githubusercontent.com/35809470/137531353-e30e4400-9821-4edb-b760-c39b1967cdf2.png)

Y Inversion:
![Screen Shot 2021-10-15 at 1 54 55 PM](https://user-images.githubusercontent.com/35809470/137531630-2c4ef7bf-cadc-44fb-a633-5e22cda5b4f9.png)

Z Inversion behaves as expected:
![Screen Shot 2021-10-15 at 1 56 10 PM](https://user-images.githubusercontent.com/35809470/137531781-c01d7b74-fb60-4926-b72c-ec7e86580537.png)

### Expected outcome

I would expect the y axis to be inverted 

### Matplotlib Version

3.2.2

### Other libraries

mpl_toolkits.mplot3d

**Comment by QuLogic** [comment]
The padding is wrong, and sometimes extra grid lines are drawn that shouldn't be. Run the tests and you will see what breaks.

**Comment by tacaswell** [comment]
There are a couple of possible gotcha's in the mplot3D code base.  The most 😱 of which is that we do some type promotion where we init the 2D version of the object and then reset the `obj.__class__` to make it the 3D version.  Another possibility is that in the `Axes3D.__init__` we may be re-setting the tick location explicitly someplace else.  A third possible thing that could be going wrong is that not all of the classes in the MRO are using `super()` in their `__init__` so the call chain is being broken at some point (either intentionally or accidentally).

My next debugging step here would be to override `set_ticks_position` like

````
def set_ticks_position(self, *args, **kwargs):
    breakpoint()
    return super().set_ticks_position(*args, **kwargs)
```

to sort out where it is being called and to add either breakpoints or prints to the new `__init__` to make sure it is _really_ being called.

**Comment by janash** [comment]
Thanks for the tip @tacaswell - it was going through the constructor, but a lot more happens after that. I added the command at a point after the axes are created, may be a bandaid fix. PR opened, lots of tests failing at the moment.

**Comment by the-kevin-jr** [comment]
Is this issue still open? I've been looking for a starting point for contribution to open source

**Comment by coding-krtek** [comment]
Hello !

So, here's what I've found about the problem that prevents you 
from fixing the ticks position.

I tried fixing it by making an __init__ method in the "YAxis"
class in "mpl_toolkits\mplot3d\axis3d.py". The problem is :
this __init__ method will be called before the _get_tick method in
the "YAxis" class in "matplotlib\axis.py". So all the changes
I make are being rewritten, and nothing changes in the end.

The only way I found to bypass this problem is by doing this :

-adding this method in "matplotlib\axis.py" -> YTick class :
    
    ...
    def fix_init(self):
        self.label1.set(verticalalignment='top',horizontalalignment='center')
        self.label2.set(verticalalignment='top',horizontalalignment='center')

-adding this line in "matplotlib\axis.py" -> YAxis class in the __init__ :

    ...
    def __init__(self, *args, **kwargs):
        self.is3d=kwargs.pop('is3d',None) # added line

-rewriting the _get_tick method in "matplotlib\axis.py" -> YAxis class :

    ...
    def _get_tick(self, major):
        if major:
            tick_kw = self._major_tick_kw
        else:
            tick_kw = self._minor_tick_kw
        my_ytick=YTick(self.axes, 0, major=major, **tick_kw)
        if self.is3d: my_ytick.fix_init()
        return my_ytick

-adding these lines in "mpl_toolkits\mplot3d\axis3d.py" -> YAxis class :

    ...
    get_data_interval, set_data_interval = maxis._make_getset_interval(
        "data", "xy_dataLim", "intervaly")
    def __init__(self,*args,**kwargs):              # added line
        super().__init__(*args,is3d=True,**kwargs)  # added line


and everything works just fine.
At least I think : I don't know how to check the safety of these addings
properly.

**Comment by janash** [comment]
@coding-krtek  - You can check your solution by running the unit tests. If you have pytest (`pip install pytest`), you can run the tests by typing `pytest` into command line when you are in your cloned repo. There are a number of tests which have reference images, and if your solution results in a difference from the stored reference images, the tests will fail. After running the tests, images from the tests will be in a folder called `result_images`

In my PR for this issue #21424, I have changed the behavior in the constructor of `Axes3D` to change the default side the y axis label is on. This solution is not yet accepted as changing axis inheritance causes changes in the reference images. For your solution did you adopt the proposed patch of fixing the classes which the 3D axes inherit from?

I can't speak for the repo maintainers, but my thought would be that you do not want to change parts of the code (for example `YAxis` and `YTick` classes) which don't have anything to do with this issue unless there is no way to avoid it.

**Comment by coding-krtek** [comment]
Um I'll be honest, I was just changing the matplotlib files directly, and I don't know how cloned repositories or virtual environments work. I tried to make a clone repo and run pytest in it as you said, but apparently "The baseline image directory does not exist".

Yes, I changed the inheritance `...(Axis, maxis.YAxis)`.
But it looks that the default 2D classes have the last word on the ticks parameters, so I don't know how to solve the problem without changing them.

**Comment by janash** [comment]
Hey @coding-krtek - if you'd like to contribute your solution to matplotlib you'll need to fork the repository and make a pull request. This way, others can see and try out your changes and they might be accepted into the main matplotlib library. You can join the gitter and ask to be put in the "incubator" channel to get help with this - https://gitter.im/matplotlib/matplotlib

**Comment by baloe** [comment]
I also noticed that `get_ticks_position()` returns `bottom` for all three axes.

**Comment by baloe** [comment]
For those in need for a workaround: I discovered that the following lines actually do invert the yaxis:

```
ylim = ax.get_ylim()
ax.set_yticks( ax.get_yticks() )
ax.set_ylim(ylim[::-1])
```

**Comment by anntzer** [comment]
Closed by #21442.
